Add support for InfluxDB 3

ngchihuan opened the pull request:
This PR introduces support for **InfluxDB 3.x** via the new `InfluxDBv3Session` class, along with integration tests and documentation updates.

ngchihuan committed:
Update pyproject.toml

ngchihuan committed:
Add influxdbv3 and tests

ngchihuan committed:
Update README.md

ngchihuan commented:
@mosmuell  Hi Möse, could you help take a look at this PR? As discussed offline, we need this new client for the InfluxDB cloud.

mosmuell commented:
Thanks for this @ngchihuan!

A couple of things:
1. Could you add the session to the default exports in `pydase_service_base/database/__init__.py`? It should look something like this:
    ```python
        try:
            import influxdb_client_3  # type: ignore # noqa
    
            from .influxdbv3_session import InfluxDBv3Session  # type: ignore
        except ImportError:
    
            class InfluxDBv3Session:  # type: ignore
                def __init__(self) -> None:
                    raise OptionalDependencyError(
                        "InfluxDBv3Session requires the 'influxdbv3' extra. "
                        "Please refer to https://gitlab.phys.ethz.ch/tiqi-projects/qchub/icon-services/pydase_service_base."
                    )
    ```
    
    and then add `"InfluxDBv3Session"` to the `__all__` list.
2. You added the `from_config_file` class method.  Do you want to instantiate the class without config files? To keep more in line with the other implementations, you could instead create a class method to instantiate it manually?
3. I don't like that you are using `session._client.query` to create a query. If you want to expose this, create a function s.t. the user does not have to access protected attributes (i.e. `_client`).
4. In the "Installation" section of the Readme, you are missing this section:
    >  - For `InfluxDBv3Session`, include the `influxdbv3` extra:
    >         ```
    >         poetry add "git+https://github.com/tiqi-group/pydase_service_base.git#main[influxdbv3]"
    >         ```
    > 
5. You have the configuration of the `influxdbv3` in the usage section. This does not align with the other databases. You could either move the config into the "Configuration" section, or remove the "Configuration" section and move all the config examples into the respective "Usage" paragraphs.
6. What are you trying to do here?
    ```python
    from influxdb_client_3 import InfluxDBClient3, Point, WritePrecision as _WritePrecision
    # ...
    
    __all__ = [
        "InfluxDBv3Session",
        "WritePrecision",
    ]
    
    WritePrecision = _WritePrecision
    ```
    Exporting the `WritePrecision` does not make sense in my eyes, and I don't understand why you would import it as `_WritePrecision` to set it to `WritePrecision` again?
7. I don't think it makes sense to have `influxdb3-python` in the `[tool.poetry.group.dev.dependencies]` section of the `pyproject.toml` file. You already have it as an extra, so if you want to run your tests, you should just install the package with the influxdbv3 extra. So I would remove it from the dev-dependencies.
8. Could you please format your files using `ruff`?

ngchihuan commented:
Thanks for reviewing @mosmuell 
1, 4, 5, 7, 8:  your suggestions make sense, will do.

2. I think it’s cleaner to keep the constructor focused on direct parameters, and move env/file-based initialization into separate helpers. That way we don’t tie object creation too tightly to config sources, and it also makes testing easier since you can just pass in values directly without having to set up env vars or config files.
One example is in the `influxdb3-client` [here](https://github.com/InfluxCommunity/influxdb3-python/blob/ed886ca8be20922f8936e94075d539fdc11d7acd/influxdb_client_3/__init__.py#L280):  

I know it is good to keep the same interface as in v1 and v2, but here I think the need of having a clear and testable interface is more important. 

3. You are right, I don't like it neither; that's why I didn't expose it :). I used it only for testing the write method. I think the test name may confuse you, so I will update the test name

6.  I think re-exporting is helpful here as I dont want our users to be concerned with influxdb_client_3 in the background.
One way to go ahead is not using alias as you suggested.

`from influxdb_client_3 import InfluxDBClient3, Point, WritePrecision`

ngchihuan committed:
Update test_influxdbv3_session.py

ngchihuan committed:
Add influxDbv3Session into database init

ngchihuan committed:
Update README.md

ngchihuan committed:
Update pyproject.toml

ngchihuan committed:
Formatting

mosmuell commented:
Regarding your comments:
2. I understand your reasoning. I feel the same way about having an explicit constructor. I suppose "breaking" the interface the other classes have does make sense. Rewriting the others might break backwards-compatibility, so I'll just keep them as-is.
3. I'm not sure what you mean by "that's why I didn't expose it": are you talking about the `query` method in general? If you just meant that you needed it for testing and you don't actually want users to use it themselves, I'm ok with that.
6. I see. I have been more on the side of using imports from external libraries directly, but I don't have a strong opinion here.

mosmuell commented:
Would you mind moving the tests into `tests/database`? I usually keep the structure of the `tests` directory aligned with the `src` directory.

ngchihuan committed:
Move test files

ngchihuan commented:
@mosmuell 
Thanks again for your comments.
I have moved the tests to `tests/database` and removed the import alias.

Regarding point 2: I don't suggest rewriting other influxdb clients. But I think it is ok to have a different interface for influxdb3 client. The influxdb3 api is totally a breaking change anyway. 

Point 3: yes, I only need it for testing. For our typical users ( physicists), I don't think they would query the data using SQL so I don't provide the query method. We have both InfluxDB and Grafana for that.

mosmuell commented:
Great, thanks!
